spec/support/saml_response_helper.rb

```diff
@@ -34,6 +34,13 @@ def raw_xml_response
       end
     end
 
+    def saml_response(settings)
+      @_saml_response ||= OneLogin::RubySaml::Response.new(
+        raw_xml_response,
+        settings: settings
+      )
+    end
```

Comment:
I find this confusing too.. I think perhaps this file need some refactoring.  Last time I tried to understand it it seemed to be working in two different ways that we could perhaps separate.